### PR TITLE
[libc][math][c23] Add {remainder,remquo}f16 C23 math functions

### PR DESCRIPTION
**Pull request by overmighty**
Part of #93566.

### CODE REVIEW

**Comment by overmighty** [comment]
cc @lntue

File: libc/src/math/generic/CMakeLists.txt [hunk 1]
```diff
@@ -2440,6 +2440,19 @@ add_entrypoint_object(
...
+  remquof16
+  SRCS
+    remquof16.cpp
+  HDRS
+    ../remquof16.h
+  DEPENDS
+    libc.src.__support.macros.properties.types
+    libc.src.__support.FPUtil.division_and_remainder_operations
+  COMPILE_OPTIONS
+    -O2
```
Comment (overmighty) [inline libc/src/math/generic/CMakeLists.txt:2453]: Should these be changed to -O3?

**Comment by overmighty** [comment]
Rebased and resolved merge conflicts.

File: libc/src/math/generic/CMakeLists.txt (hunk 1 above)
Comment (lntue) [reply libc/src/math/generic/CMakeLists.txt:2547]: Yes, please change all of them to -O3

**Review by lntue — APPROVED** [review]

**Pull request merged**